### PR TITLE
fix: BasicAuthAuthenticationFetcher should use TokenConfiguration

### PR DESCRIPTION
**Pull request by sdelamo**
Closes bug: https://github.com/micronaut-projects/micronaut-security/issues/284

We should always use `TokenConfiguration::getRolesNames`. 

Close: https://github.com/micronaut-projects/micronaut-security/issues/286 as well.

### CODE REVIEW

File: security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationFetcher.java [hunk 1]
```diff
@@ -65,9 +67,9 @@ public BasicAuthAuthenticationFetcher(Authenticator authenticator,
             Flowable<AuthenticationResponse> authenticationResponse = Flowable.fromPublisher(authenticator.authenticate(request, credentials.get()));
 
             return authenticationResponse.switchMap(response -> {
-                if (response.isAuthenticated()) {
+                if (response.isAuthenticated() && response.getUserDetails().isPresent()) {
```
Comment (jameskleeh) [inline security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationFetcher.java:70]: The contract of the API is that if `isAuthenticated` returns true, the user details should be present. So this check is unnecessary

File: security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationFetcher.java [hunk 2]
```diff
@@ -36,21 +38,21 @@
...
-    private final BasicAuthAuthenticationConfiguration configuration;
+    private final TokenConfiguration configuration;
 
     /**
      * @param authenticator The authenticator to authenticate the credentials
      * @param configuration The basic authentication configuration
      */
     public BasicAuthAuthenticationFetcher(Authenticator authenticator,
-                                          BasicAuthAuthenticationConfiguration configuration) {
+                                          TokenConfiguration configuration) {
```
Comment (jameskleeh) [inline security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationFetcher.java:55]: I'm not sure I agree we should be using token configuration here. A basic auth string is not a token, so it doesn't make sense for users to configure it via ...token..

File: security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationConfiguration.java [hunk 3]
```diff
@@ -1,70 +0,0 @@
...
-import io.micronaut.security.config.SecurityConfigurationProperties;
-
-/**
- * Configuration for basic authentication.
- *
- * @author James Kleeh
- * @since 2.0.0
- */
-@ConfigurationProperties(BasicAuthAuthenticationConfiguration.PREFIX)
-public class BasicAuthAuthenticationConfiguration implements Toggleable {
```
Comment (jameskleeh) [inline security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationConfiguration.java:29]: I don't think this should be deleted for documentation purposes

File: security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationFetcher.java (hunk 2 above)
Comment (sdelamo) [reply security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationFetcher.java:55]: Maybe the issue is `TokenConfiguration` should be renamed to `AuthenticationConfiguration`.

File: security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationFetcher.java (hunk 1 above)
Comment (sdelamo) [reply security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationFetcher.java]: I've removed the check.

File: security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationConfiguration.java (hunk 3 above)
Comment (sdelamo) [reply security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationConfiguration.java:29]: I've documented in the docs how to disable. I am not confortable leaving a class just for the `Toggleable` field. Moreover, it was never displayed in the main docs. Only in configuration reference.

**Comment by sdelamo** [comment]
@jameskleeh what do you think? Do you want me to rename `TokenConfiguration` to `AuthenticationConfiguration`?

File: security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationFetcher.java (hunk 2 above)
Comment (sdelamo) [reply security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationFetcher.java:55]: Please note that the reason for this pr is that not using `TokenConfiguration::getRolesName` cuases a bug. See steps the reproduce here: https://github.com/micronaut-projects/micronaut-security/issues/284

**Pull request merged**